tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -261,6 +279,23 @@ def test_download_and_prepare_as_dataset(self):
     else:
       self._download_and_prepare_as_dataset(self.builder)
 
+    if not self.SKIP_CHECKSUMS:
+      with self._subTest("url_checksums"):
+        self._test_checksums()
+
+  def _test_checksums(self):
+    urls = []
+    # how to get the path to the checksum file corresponding to the testcase.
+    filepath = os.path.join(download.checksums._get_path(os.path.basename(self.example_dir)))
+    if tf.io.gfile.exists(filepath):
+      with tf.io.gfile.GFile(filepath, "r") as f:
+        for line in f.readlines():
+          urls.append(line.split()[0])
+    else:
+      raise AssertionError("url checksums file not found at %s" % filepath)
+
+    self.assertTrue(self._download_urls.issubset(set(urls)), "url checksums don't match.")
```

Comment:
All URLs are not encountered. So `_download_urls` is only a subset of all URLs in the url_checksum file.
Please correct me, if I am wrong.

Reply:
Yes, this is correct (ex:for builder config)